page-tracking/src/page_tracker.rs

```diff
@@ -298,6 +324,55 @@ impl PageTracker {
         }
     }
 
+    /// Acquires an exclusive reference to the shareable page at `addr` if it's owned by owner,
+    /// and in Mapped or Shared state
+    pub fn get_shareable_page<P: PhysPage>(
+        &self,
+        addr: SupervisorPageAddr,
+        owner: PageOwnerId,
+    ) -> Result<P> {
+        let mut page_tracker = self.inner.lock();
+        let info = page_tracker.get_mut(addr)?;
+        if info.owner() != Some(owner) || info.mem_type() != P::mem_type() {
+            return Err(Error::PageNotShareable);
+        }
+
+        match info.state() {
+            // Ensure that the share RC can be incremented before attempting to obtain the lock.
+            PageState::Shared(rc) if rc == u64::MAX => {
+                return Err(Error::PageNotShareable);
+            }
+            PageState::Shared(_) | PageState::Mapped => {}
+            _ => {
+                return Err(Error::PageNotShareable);
+            }
+        };
```

Comment:
nit: You could put this in a `PageInfo::is_shareable()` helper, similar in spirit to the existing `is_convertible()` helper.